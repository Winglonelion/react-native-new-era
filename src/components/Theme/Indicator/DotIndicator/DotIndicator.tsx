import React, { memo } from 'react';
import { StyleSheet, View } from 'react-native';
import Colors from 'theme/colors';

interface PropTypes {
  isActive?: boolean;
}

const DotIndicator: React.FC<PropTypes> = memo(({ isActive = false }) => {
  return <View style={[styles.dot, isActive && styles.activeDot]} />;
});

const styles = StyleSheet.create({
  dot: {
    backgroundColor: Colors.grey,
    width: 8,
    height: 8,
    borderRadius: 8,
    marginHorizontal: 6,
  },
  activeDot: {
    backgroundColor: Colors.black,
  },
});

export default DotIndicator;
