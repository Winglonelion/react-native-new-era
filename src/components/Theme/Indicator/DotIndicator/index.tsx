export { default as default } from './DotIndicator';
