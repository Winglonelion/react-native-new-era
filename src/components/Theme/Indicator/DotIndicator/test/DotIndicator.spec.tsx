import React from 'react';
import { shallow } from 'enzyme';
import DotIndicator, { PropTypes } from '../DotIndicator';

const setup = (propOverrides: PropTypes = {} as PropTypes) => {
  const props = { ...propOverrides };
  const wrapper = shallow(<DotIndicator {...props} />);
  return {
    props,
    wrapper,
  };
};

describe('DotIndicator check snapshot', () => {
  test('should match snapshot with isActive true', () => {
    const { wrapper } = setup({ isActive: true });
    expect(wrapper).toMatchSnapshot();
  });

  test('should match snapshot with isActive false', () => {
    const { wrapper } = setup({ isActive: false });
    expect(wrapper).toMatchSnapshot();
  });
});
