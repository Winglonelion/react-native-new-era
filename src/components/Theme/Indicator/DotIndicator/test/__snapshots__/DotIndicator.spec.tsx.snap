// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DotIndicator check snapshot should match snapshot with isActive false 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#DADFE6",
        "borderRadius": 8,
        "height": 8,
        "marginHorizontal": 6,
        "width": 8,
      },
      false,
    ]
  }
/>
`;

exports[`DotIndicator check snapshot should match snapshot with isActive true 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#DADFE6",
        "borderRadius": 8,
        "height": 8,
        "marginHorizontal": 6,
        "width": 8,
      },
      Object {
        "backgroundColor": "#000000",
      },
    ]
  }
/>
`;
